src/main/java/graphql/execution/DataFetcherResult.java

```diff
@@ -0,0 +1,40 @@
+package graphql.execution;
+
+import graphql.GraphQLError;
+import graphql.schema.DataFetcher;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+
+/**
+ * An object that can be returned from a {@link DataFetcher} that contains both data and errors to be relativized and
+ * added to the final result.
+ *
+ * @param <T> The type of the data fetched
+ */
+public class DataFetcherResult<T> {
```

Comment:
Could also be an interface; just thought this way it'd be more user-friendly.

Reply:
class is fine if we want Users to directly instantiate it.